astropy/cosmology/flrw/w0wzcdm.py

```diff
@@ -203,3 +203,124 @@ def de_density_scale(self, z):
+        Derivative of the dark energy equation of state with respect to z.
+        A cosmological constant has w0=-1.0 and wz=0.0.
+
+    Tcmb0 : float or scalar quantity-like ['temperature'], optional
+        Temperature of the CMB z=0. If a float, must be in [K]. Default: 0 [K].
+        Setting this to zero will turn off both photons and neutrinos
+        (even massive ones).
+
+    Neff : float, optional
+        Effective number of Neutrino species. Default 3.04.
+
+    m_nu : quantity-like ['energy', 'mass'] or array-like, optional
+        Mass of each neutrino species in [eV] (mass-energy equivalency enabled).
+        If this is a scalar Quantity, then all neutrino species are assumed to
+        have that mass. Otherwise, the mass of each species. The actual number
+        of neutrino species (and hence the number of elements of m_nu if it is
+        not scalar) must be the floor of Neff. Typically this means you should
+        provide three neutrino masses unless you are considering something like
+        a sterile neutrino.
+
+    Ob0 : float or None, optional
+        Omega baryons: density of baryonic matter in units of the critical
+        density at z=0.  If this is set to None (the default), any computation
+        that requires its value will raise an exception.
+
+    name : str or None (optional, keyword-only)
+        Name for this cosmological object.
+
+    meta : mapping or None (optional, keyword-only)
+        Metadata for the cosmology, e.g., a reference.
+
+    Examples
+    --------
+    >>> from astropy.cosmology import Flatw0wzCDM
+    >>> cosmo = Flatw0wzCDM(H0=70, Om0=0.3, w0=-0.9, wz=0.2)
+
+    The comoving distance in Mpc at redshift z:
+
+    >>> cosmo.comoving_distance(0.5)
+    <Quantity 1982.66012926 Mpc>
+    """
+
+    def __init__(
+        self,
+        H0,
+        Om0,
+        w0=-1.0,
+        wz=0.0,
+        Tcmb0=0.0 * u.K,
+        Neff=3.04,
+        m_nu=0.0 * u.eV,
+        Ob0=None,
+        *,
+        name=None,
+        meta=None
+    ):
+        super().__init__(
+            H0=H0,
+            Om0=Om0,
+            Ode0=0.0,
+            w0=w0,
+            wz=wz,
+            Tcmb0=Tcmb0,
+            Neff=Neff,
+            m_nu=m_nu,
+            Ob0=Ob0,
+            name=name,
+            meta=meta,
+        )
+
+        # Please see :ref:`astropy-cosmology-fast-integrals` for discussion
+        # about what is being done here.
+        if self._Tcmb0.value == 0:
+            self._inv_efunc_scalar = scalar_inv_efuncs.fw0wzcdm_inv_efunc_norel
+            self._inv_efunc_scalar_args = (self._Om0, self._Ode0, self._w0, self._wz)
+        elif not self._massivenu:
+            self._inv_efunc_scalar = scalar_inv_efuncs.fw0wzcdm_inv_efunc_nomnu
+            self._inv_efunc_scalar_args = (
+                self._Om0,
+                self._Ode0,
+                self._Ogamma0 + self._Onu0,
+                self._w0,
+                self._wz,
+            )
```

Comment:
Setting up a dictionary that collects the kwargs can help reduce code duplication.
```suggestion
        self.inv_efunc_scalar_kwargs = {
            "Om0": self._Om0, "Ode0": self._Ode0, "w0": self._w0, "wz": self._wz
        }
        if self._Tcmb0.value == 0:
            self._inv_efunc_scalar = scalar_inv_efuncs.fw0wzcdm_inv_efunc_norel
        elif not self._massivenu:
            self._inv_efunc_scalar = scalar_inv_efuncs.fw0wzcdm_inv_efunc_nomnu
            self._inv_efunc_scalar_kargs["Ogamma0"] = self._Ogamma0 + self._Onu0
        else:
            ...
```
 (although there's a good chance I got the names of the keyword arguments wrong)

Reply:
While this would be nice, for the c code these need to be positional arguments.

Reply:
How about setting up the list `[self._Om0, self._Ode0, self._w0, self._wz]` that would be extended in the `elif` and `else` cases? It would require changing the order of some of the arguments in the Cython function signatures, but if they are not public then that can be done.

Reply:
I like this idea! Let's open a new issue to do this in a followup PR for all the cosmologies at once.

Reply:
I'm not too familiar with `astropy.cosmology`. If you think a separate refactoring pull request is worth it then let's do that.

Reply:
Yeah. It think it's a good idea to do all these in one go. And investigate if anyone is using this in the wild. Or if dictionary access is as performant and change it from positional args.
If course the largest benefit would be to change this from an quad integral to an ODE. Then evaluations can be calculated summatively, not individually.